
# HG changeset patch
# User Ullrich Koethe <ullrich.koethe@iwr.uni-heidelberg.de>
# Date 1299851742 -3600
# Node ID c3a55cf81b34ae5e71e14b1dc90ece7e646a02dd
# Parent  90494fd71974b95c61181a47580f4d1f02642723
update for libpng 1.5 (fix by Mark Brand, mingw-cross-env)

diff -r 90494fd71974 -r c3a55cf81b34 src/impex/png.cxx
--- a/src/impex/png.cxx	Thu Mar 10 18:47:50 2011 +0100
+++ b/src/impex/png.cxx	Fri Mar 11 14:55:42 2011 +0100
@@ -320,7 +320,11 @@
 #if (PNG_LIBPNG_VER > 10008) && defined(PNG_READ_iCCP_SUPPORTED)
         char * dummyName;
         int dummyCompType;
+#if (PNG_LIBPNG_VER < 10500)
         char * profilePtr;
+#else
+        png_byte * profilePtr;
+#endif
         png_uint_32 profileLen;
         if (png_get_valid( png, info, PNG_INFO_iCCP )) {
             png_get_iCCP(png, info, &dummyName, &dummyCompType, &profilePtr, &profileLen) ;
@@ -593,7 +597,11 @@
         // set icc profile
         if (iccProfile.size() > 0) {
             png_set_iCCP(png, info, (png_charp)("icc"), 0,
+#if (PNG_LIBPNG_VER < 10500)
                          (png_charp)iccProfile.begin(), (png_uint_32)iccProfile.size());
+#else
+                         (png_byte*)iccProfile.begin(), (png_uint_32)iccProfile.size());
+#endif
         }
 #endif
 

